### ISSUE TITLE
First dialog does not receive focus on OS X

### ISSUE THREAD

**Issue by ArekSredzki** [atom/electron#4719]
* Electron version: 0.36.10
* Operating system: OS X 10.11.3

Dialog boxes should be given focus on-open as they will hide behind the current window in certain cases on OS X.
- Dialog box is opened before renderer windows (ex. directory/file selection)
  - If opened from command line.
  - Opened from finder
  - (unconfirmed but probable) All scenarios

**Comment by kevinsawicki** [comment]
Could you provide a small sample app (or just a `main.js`) that demonstrates this issue? Thanks.

**Comment by ArekSredzki** [comment]
@kevinsawicki Gladly

```js
'use strict';

const electron = require('electron');
// Module to control application life.
const app = electron.app;
const dialog = electron.dialog;

app.on('ready', function () {
  dialog.showMessageBox({
          type: 'question',
          message: 'Do you see this dialog as focused on start?',
          detail: 'This is the best application you have ever used, get ready.',
          buttons: ['I am not on OS X', 'No'],
          cancelId: 1,
          defaultId: 0
        }, function() {
          app.quit();
        });
});
```

**Comment by phamdaniel** [comment]
This is happening because alerts and dialogs currently don't force the app to get focused, which i think is the for the best.
If you want the dialog to be the first thing that's rendered, I think you should probably call `app.focus()` before displaying it.

**Comment by ArekSredzki** [comment]
@phamdaniel Thank you for clarifying that this is the intended action. However, app does not offer a focus method, only browser windows do (at least according to the documentation).

**Comment by phamdaniel** [comment]
It's there alright, just not documented. I submitted a PR to take care of it.

**Comment by ArekSredzki** [comment]
Thanks :+1:

**Comment by zcbenz** [comment]
Closed with #4725.
